src/pages/LoginSignup/components/Input.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect } from 'react';
+import './input.scss';
+
+const Input = ({ name, type, text, value, handleInput, handleSignUpValid }) => {
+  const [isValid, setIsValid] = useState(false);
+
+  const valid = value => {
+    const validTest = name === 'signUp' && handleSignUpValid(value);
+    setIsValid(validTest);
+  };
+
+  useEffect(() => {
+    valid(value);
+  });
```

Comment:
```suggestion
  useEffect(() => {
    valid(value);
  }, [value]);
```

value가 바뀔 때에만 유효성검사를 할 필요가 있을 것 같으니, value를 의존성배열에 명확히 넣어주는게 좋아보이네요! 안그러면 다른 props이나 state가 바껴서 re-render될 때도 매번 불필요하게 valid함수를 호출합니다!


그리고 가장 좋은 방법은 굳이 valid를 state로 안하고 그냥 일반 변수로 선언해도 하는게 좋아보이네요!

다른 props으로부터 계산 가능한 값은 굳이 다시 state로 만들필요가 없습니다.

[공식문서 React로 사고하기](https://ko.reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state) 를 보시면 어떤 값들이 state가 되어야하는지에 대해 적혀있습니다.

> 각각 살펴보고 어떤 게 state가 되어야 하는 지 살펴봅시다. 이는 각 데이터에 대해 아래의 세 가지 질문을 통해 결정할 수 있습니다
> 

> 1. 부모로부터 props를 통해 전달됩니까? 그러면 확실히 state가 아닙니다.
> 

> 2. 시간이 지나도 변하지 않나요? 그러면 확실히 state가 아닙니다.
> 

> 3. 컴포넌트 안의 다른 state나 props를 가지고 계산 가능한가요? 그렇다면 state가 아닙니다.
>

Reply:
일반 변수로 선언하는 방법으로 변경했습니다. 불필요한 Hook을 두가지나 줄일수 있었어요